2021-04-22 23:20:06,226 INFO werkzeug  * Running on http://localhost:5000/ (Press CTRL+C to quit)
2021-04-22 23:20:10,213 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:20:10] "[37mGET /users/betters/0 HTTP/1.1[0m" 200 -
2021-04-22 23:20:10,317 CRITICAL root Not Found
2021-04-22 23:20:10,317 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:20:10] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:20:11,680 INFO root Info about users in 1 day was received
2021-04-22 23:20:11,694 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:20:11] "[37mGET /1 HTTP/1.1[0m" 200 -
2021-04-22 23:20:12,227 CRITICAL root Not Found
2021-04-22 23:20:12,227 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:20:12] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:20:12,683 INFO root Info about users in 0 day was received
2021-04-22 23:20:12,697 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:20:12] "[37mGET /0 HTTP/1.1[0m" 200 -
2021-04-22 23:20:13,233 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:20:13] "[37mGET /users/betters/0 HTTP/1.1[0m" 200 -
2021-04-22 23:20:13,341 CRITICAL root Not Found
2021-04-22 23:20:13,342 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:20:13] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:21:25,593 INFO werkzeug  * Running on http://localhost:5000/ (Press CTRL+C to quit)
2021-04-22 23:21:28,436 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:21:28] "[37mGET /users/betters/0 HTTP/1.1[0m" 200 -
2021-04-22 23:21:28,534 CRITICAL root Not Found
2021-04-22 23:21:28,534 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:21:28] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:21:50,289 INFO werkzeug  * Running on http://localhost:5000/ (Press CTRL+C to quit)
2021-04-22 23:21:53,648 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:21:53] "[37mGET /users/betters/0 HTTP/1.1[0m" 200 -
2021-04-22 23:21:53,738 CRITICAL root Not Found
2021-04-22 23:21:53,739 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:21:53] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:21:54,742 INFO root Info about users in 0 day was received
2021-04-22 23:21:54,756 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:21:54] "[37mGET /0 HTTP/1.1[0m" 200 -
2021-04-22 23:21:55,283 CRITICAL root Not Found
2021-04-22 23:21:55,283 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:21:55] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:25:47,191 INFO root Info about users in 0 day was received
2021-04-22 23:25:47,193 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:25:47] "[37mGET /0?subject=История&class=5 HTTP/1.1[0m" 200 -
2021-04-22 23:25:47,304 CRITICAL root Not Found
2021-04-22 23:25:47,305 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:25:47] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:25:52,185 INFO root Info about users in 0 day was received
2021-04-22 23:25:52,187 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:25:52] "[37mGET /0?subject=История&class= HTTP/1.1[0m" 200 -
2021-04-22 23:25:52,492 CRITICAL root Not Found
2021-04-22 23:25:52,493 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:25:52] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:27:40,289 INFO werkzeug  * Running on http://localhost:5000/ (Press CTRL+C to quit)
2021-04-22 23:27:52,783 INFO root Info about users in 0 day was received
2021-04-22 23:27:52,802 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:27:52] "[37mGET /0?subject=История&class= HTTP/1.1[0m" 200 -
2021-04-22 23:27:52,892 CRITICAL root Not Found
2021-04-22 23:27:52,893 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:27:52] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:27:57,344 INFO root Info about users in 0 day was received
2021-04-22 23:27:57,357 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:27:57] "[37mGET /0 HTTP/1.1[0m" 200 -
2021-04-22 23:27:57,895 CRITICAL root Not Found
2021-04-22 23:27:57,895 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:27:57] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:28:18,256 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:18] "[32mGET /logout HTTP/1.1[0m" 302 -
2021-04-22 23:28:18,265 INFO root Info about users in 0 day was received
2021-04-22 23:28:18,276 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:18] "[37mGET / HTTP/1.1[0m" 200 -
2021-04-22 23:28:18,817 CRITICAL root Not Found
2021-04-22 23:28:18,817 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:18] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:28:23,162 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:23] "[37mGET /login HTTP/1.1[0m" 200 -
2021-04-22 23:28:23,216 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:23] "[36mGET /static/images/univers.png HTTP/1.1[0m" 304 -
2021-04-22 23:28:23,424 CRITICAL root Not Found
2021-04-22 23:28:23,424 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:23] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:28:26,331 INFO root Admin №5 Тестер with subject ИЦН and email 2004maydurov@mail.ru was connected to site
2021-04-22 23:28:26,421 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:26] "[32mPOST /login HTTP/1.1[0m" 302 -
2021-04-22 23:28:26,436 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:26] "[37mGET /admin HTTP/1.1[0m" 200 -
2021-04-22 23:28:26,591 CRITICAL root Not Found
2021-04-22 23:28:26,592 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:26] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:28:31,859 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:31] "[32mGET /logout HTTP/1.1[0m" 302 -
2021-04-22 23:28:31,868 INFO root Info about users in 0 day was received
2021-04-22 23:28:31,878 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:31] "[37mGET / HTTP/1.1[0m" 200 -
2021-04-22 23:28:32,405 CRITICAL root Not Found
2021-04-22 23:28:32,406 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:32] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:28:32,935 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:32] "[37mGET /login HTTP/1.1[0m" 200 -
2021-04-22 23:28:33,064 CRITICAL root Not Found
2021-04-22 23:28:33,065 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:33] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:28:35,858 INFO root Admin №5 Тестер with subject ИЦН and email 2004maydurov@mail.ru was connected to site
2021-04-22 23:28:35,953 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:35] "[32mPOST /login HTTP/1.1[0m" 302 -
2021-04-22 23:28:36,164 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:36] "[37mGET /admin HTTP/1.1[0m" 200 -
2021-04-22 23:28:36,564 CRITICAL root Not Found
2021-04-22 23:28:36,565 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:36] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:28:37,864 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:37] "[32mGET /logout HTTP/1.1[0m" 302 -
2021-04-22 23:28:38,176 INFO root Info about users in 0 day was received
2021-04-22 23:28:38,187 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:38] "[37mGET / HTTP/1.1[0m" 200 -
2021-04-22 23:28:39,024 CRITICAL root Not Found
2021-04-22 23:28:39,024 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:39] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:28:39,606 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:39] "[37mGET /login HTTP/1.1[0m" 200 -
2021-04-22 23:28:39,914 CRITICAL root Not Found
2021-04-22 23:28:39,914 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:39] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:28:43,794 INFO root Admin №3 Майдуров Артём Игоревич with subject Главный админ and email b@mail.ru was connected to site
2021-04-22 23:28:43,880 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:43] "[32mPOST /login HTTP/1.1[0m" 302 -
2021-04-22 23:28:43,887 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:43] "[37mGET /admin HTTP/1.1[0m" 200 -
2021-04-22 23:28:44,044 CRITICAL root Not Found
2021-04-22 23:28:44,045 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:28:44] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:29:15,310 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:29:15] "[32mGET /users/change-day/1 HTTP/1.1[0m" 302 -
2021-04-22 23:29:15,317 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:29:15] "[37mGET /admin HTTP/1.1[0m" 200 -
2021-04-22 23:29:15,559 CRITICAL root Not Found
2021-04-22 23:29:15,560 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:29:15] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:29:16,728 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:29:16] "[32mGET /users/change-day/2 HTTP/1.1[0m" 302 -
2021-04-22 23:29:16,735 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:29:16] "[37mGET /admin HTTP/1.1[0m" 200 -
2021-04-22 23:29:16,989 CRITICAL root Not Found
2021-04-22 23:29:16,990 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:29:16] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:29:18,053 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:29:18] "[32mGET /logout HTTP/1.1[0m" 302 -
2021-04-22 23:29:18,062 INFO root Info about users in 0 day was received
2021-04-22 23:29:18,074 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:29:18] "[37mGET / HTTP/1.1[0m" 200 -
2021-04-22 23:29:18,601 CRITICAL root Not Found
2021-04-22 23:29:18,601 INFO werkzeug 127.0.0.1 - - [22/Apr/2021 23:29:18] "[37mGET /favicon.ico HTTP/1.1[0m" 200 -
2021-04-22 23:45:06,300 INFO waitress Serving on http://[::1]:8080
2021-04-22 23:45:06,300 INFO waitress Serving on http://127.0.0.1:8080
2021-04-22 23:45:30,907 INFO root Info about users in 0 day was received
2021-04-22 23:45:31,549 CRITICAL root Not Found
2021-04-23 00:06:25,080 INFO waitress Serving on http://[::1]:8080
2021-04-23 00:06:25,081 INFO waitress Serving on http://127.0.0.1:8080
2021-04-23 00:06:41,438 INFO root Info about users in 0 day was received
2021-04-23 00:06:41,457 ERROR main_2 Exception on / [GET]
Traceback (most recent call last):
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 85, in main
    return render_template("main.html", day=1, users=data, config=config, subjects=list(subjects))
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\templating.py", line 137, in render_template
    return _render(
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\templating.py", line 120, in _render
    rv = template.render(context)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\jinja2\environment.py", line 1090, in render
    self.environment.handle_exception()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\jinja2\environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\jinja2\_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "D:\Projects\PyCharm\FlaskWeb\templates\main.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "D:\Projects\PyCharm\FlaskWeb\templates\base.html", line 7, in top-level template code
    <link rel="shortcut icon" href="{{ url_for("static", "images/univers.png") }}">
TypeError: url_for() takes 1 positional argument but 2 were given
2021-04-23 00:06:41,460 CRITICAL root Internal Server Error
2021-04-23 00:06:41,573 CRITICAL root Not Found
2021-04-23 00:07:09,225 INFO waitress Serving on http://[::1]:8080
2021-04-23 00:07:09,225 INFO waitress Serving on http://127.0.0.1:8080
2021-04-23 00:07:10,807 INFO root Info about users in 0 day was received
2021-04-23 11:24:26,692 INFO waitress Serving on http://[::1]:8080
2021-04-23 11:24:26,696 INFO waitress Serving on http://127.0.0.1:8080
2021-04-23 11:24:48,637 INFO root Info about users in 0 day was received
2021-04-23 11:25:14,592 ERROR main_2 Exception on /users/betters/0 [GET]
Traceback (most recent call last):
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 262, in betters_students
    return render_template("main.html", day=day + 1, users=filtered_data(args, students)[:20], subjects=subjects)
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 140, in filtered_data
    data = list(filter(lambda x: value in x["results"].keys(), data))
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 140, in <lambda>
    data = list(filter(lambda x: value in x["results"].keys(), data))
KeyError: 'results'
2021-04-23 11:25:14,630 CRITICAL root Internal Server Error
2021-04-23 11:25:14,716 CRITICAL root Not Found
2021-04-23 11:28:07,552 INFO waitress Serving on http://[::1]:8080
2021-04-23 11:28:07,553 INFO waitress Serving on http://127.0.0.1:8080
2021-04-23 11:30:14,061 INFO waitress Serving on http://[::1]:8080
2021-04-23 11:30:14,061 INFO waitress Serving on http://127.0.0.1:8080
2021-04-23 11:30:25,798 ERROR main_2 Exception on /users/betters/0 [GET]
Traceback (most recent call last):
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 259, in betters_students
    students = sorted(d["users"].copy(), reverse=True,
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 261, in <lambda>
    sorted(x["days"][day].values(), reverse=True, key=lambda j: j[1])[:2])))
KeyError: 'days'
2021-04-23 11:30:25,800 CRITICAL root Internal Server Error
2021-04-23 11:30:25,873 CRITICAL root Not Found
2021-04-23 11:31:11,317 INFO waitress Serving on http://[::1]:8080
2021-04-23 11:31:11,317 INFO waitress Serving on http://127.0.0.1:8080
2021-04-23 11:31:29,611 INFO root Info about users in 0 day was received
2021-04-23 11:32:04,347 INFO waitress Serving on http://[::1]:8080
2021-04-23 11:32:04,347 INFO waitress Serving on http://127.0.0.1:8080
2021-04-23 11:32:09,889 ERROR main_2 Exception on /users/betters/1 [GET]
Traceback (most recent call last):
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 259, in betters_students
    students = sorted(d["users"].copy(), reverse=True,
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 261, in <lambda>
    sorted(x["days"][day].values(), reverse=True, key=lambda j: j[1])[:2])))
KeyError: 'days'
2021-04-23 11:32:09,891 CRITICAL root Internal Server Error
2021-04-23 11:32:09,949 CRITICAL root Not Found
2021-04-23 11:33:14,008 INFO waitress Serving on http://[::1]:8080
2021-04-23 11:33:14,008 INFO waitress Serving on http://127.0.0.1:8080
2021-04-23 11:33:19,794 ERROR main_2 Exception on /users/betters/1 [GET]
Traceback (most recent call last):
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 259, in betters_students
    students = sorted(d["users"].copy(), reverse=True,
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 261, in <lambda>
    sorted(x["days"][day].values(), reverse=True, key=lambda j: j[1])[:2])))
KeyError: 'days'
2021-04-23 11:33:19,796 CRITICAL root Internal Server Error
2021-04-23 11:33:19,851 CRITICAL root Not Found
2021-04-23 11:33:21,673 ERROR main_2 Exception on /users/betters/1 [GET]
Traceback (most recent call last):
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 259, in betters_students
    students = sorted(d["users"].copy(), reverse=True,
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 261, in <lambda>
    sorted(x["days"][day].values(), reverse=True, key=lambda j: j[1])[:2])))
KeyError: 'days'
2021-04-23 11:33:21,674 CRITICAL root Internal Server Error
2021-04-23 11:33:21,727 CRITICAL root Not Found
2021-04-23 11:33:30,339 INFO waitress Serving on http://[::1]:8080
2021-04-23 11:33:30,339 INFO waitress Serving on http://127.0.0.1:8080
2021-04-23 11:34:04,948 ERROR main_2 Exception on /users/betters/0 [GET]
Traceback (most recent call last):
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "D:\Projects\PyCharm\FlaskWeb\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 259, in betters_students
    students = sorted(d["users"].copy(), reverse=True,
  File "D:\Projects\PyCharm\FlaskWeb\main_2.py", line 261, in <lambda>
    sorted(x["days"][day].values(), reverse=True, key=lambda j: j[1])[:2])))
KeyError: 'days'
2021-04-23 11:34:04,950 CRITICAL root Internal Server Error
2021-04-23 11:34:05,006 CRITICAL root Not Found
